docs/BreakingChanges/100 Foreach iterator variable is now scoped within the iteration.md

```diff
@@ -0,0 +1,22 @@
+## 100: Foreach iterator variable is now scoped within the iteration, so closure capturing semantics are different (in C#5)
```

Comment:
{
  "Id": "100",
  "Title": "Foreach iterator variable is now scoped within the iteration, so closure capturing semantics are different (in C#5)",
  "Details": "Beginning with C# 5 (Visual Studio 2012), foreach iterator variables are scoped within the iteration. This can cause breaks if code was previously depending on the variables to not be included in the foreach's closure. The symptom of this change will be that an iterator variable passed to a delagate will be treated as the value it had at the time the delegate was created, rather than the value it had at the time the delegate was invoked.",
  "Suggestion": "Ideally, code should be updated to expect the new compiler behavior. If the old semantics are required, the iterator variable can be replaced with a separate variable which is explicitly placed outside of the loop's scope.",
  "Link": "http://blogs.msdn.com/b/ericlippert/archive/2009/11/12/closing-over-the-loop-variable-considered-harmful.aspx",
  "Markdown": null,
  "ApplicableApis": [
    "Not detectable via API analysis"
  ],
  "Related": null,
  "VersionBroken": "4.5",
  "VersionFixed": null,
  "IsBuildTime": true,
  "IsQuirked": false,
  "IsSourceAnalyzerAvailable": true,
  "BugLink": null,
  "Notes": null,
  "IsRetargeting": true,
  "ImpactScope": 1
}

**Parsed APIs:**
- [ ] Not detectable via API analysis